Source/Core/VideoBackends/D3D12/D3DShader.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright 2010 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <string>
+
+#include "Common/StringUtil.h"
+#include "VideoBackends/D3D12/D3DBase.h"
+#include "VideoBackends/D3D12/D3DShader.h"
+#include "VideoCommon/VideoConfig.h"
+
+namespace DX12
+{
+
+namespace D3D
+{
+
+// code->bytecode
+bool CompileVertexShader(const std::string& code, D3DBlob** blob)
```

Comment:
Please merge those functions CompileVertexShader, CompileGeometryShader and CompilePixelShader.